moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp

```diff
@@ -75,23 +76,23 @@ RobotModelLoader::~RobotModelLoader()
 
 namespace
 {
-bool canSpecifyPosition(const moveit::core::JointModel* jmodel, const unsigned int index)
+bool canSpecifyPosition(const moveit::core::JointModel* jmodel, const unsigned int index, const rclcpp::Logger& logger)
```

Comment:
ouch

Reply:
this is one of the trade-offs of carrying around state everywhere for logging. for the child loggers you have to pass them around (either by putting them in shared state in a class or by using function parameters).

Reply:
For free functions that are public I think we should just use the global node logger and users can use `{function}` to filter for those logs.

For functions like this one that are in the private namespace of cpp file, adding parameters to it has no affect on the public api and seems fine to me.